pallets/protos/src/lib.rs

```diff
@@ -418,6 +420,8 @@ pub mod pallet {
 
 			ensure!(!data.is_empty(), Error::<T>::ProtoDataIsEmpty);
 
+			ensure!(!tags.iter().enumerate().any(|(index, tag)| tags.iter().enumerate().any(|(i, t)| t == tag && i != index)), Error::<T>::DuplicateProtoTagExists); // TODO Review - Is `O(n ^ 2)` good? (Alternatively we can **use HashMap** or **sort the tags then check for equal consecutive elements** -  but I don't think it's worth it since `T::MaxTags` is small
```

Comment:
it's fine since t(max) is predictable

Reply:
is this reason okay? - https://github.com/fragcolor-xyz/clamor/pull/238#discussion_r1045316357 
@sinkingsugar